pkg/selector/outputs/bubbletea.go

```diff
@@ -257,3 +326,76 @@ func createTable(instanceTypes []*instancetypes.Details) table.Model {
 
 	return newTable
 }
+
+// resizeTableView will change the dimensions of the table in order to accommodate
+// the new window dimensions represented by the given tea.WindowSizeMsg
+func resizeTableView(model BubbleTeaModel, msg tea.WindowSizeMsg) BubbleTeaModel {
+	// This is needed to handle a bug with bubble tea
+	// where resizing causes misprints (https://github.com/Evertras/bubble-table/issues/121)
+	termenv.ClearScreen()
+
+	// handle width changes
+	model.TableModel = model.TableModel.WithMaxTotalWidth(msg.Width)
+
+	// handle height changes
+	if headerAndFooterPadding >= msg.Height {
+		// height too short to fit rows
+		model.TableModel = model.TableModel.WithPageSize(0)
+	} else {
+		newRowsPerPage := msg.Height - headerAndFooterPadding
+		model.TableModel = model.TableModel.WithPageSize(newRowsPerPage)
+	}
+
+	return model
+}
+
+// verbose helpers:
+
+// styling for viewport
+var (
+	titleStyle = func() lipgloss.Style {
+		b := lipgloss.RoundedBorder()
+		b.Right = "├"
+		return lipgloss.NewStyle().BorderStyle(b).Padding(0, 1)
+	}()
+
+	infoStyle = func() lipgloss.Style {
+		b := lipgloss.RoundedBorder()
+		b.Left = "┤"
+		return titleStyle.Copy().BorderStyle(b)
+	}()
+)
+
+// initVerboseView initializes and returns a new verboseState based on the given
+// instance type details
+func initVerboseView(instanceTypes []*instancetypes.Details) *verboseState {
+	viewportModel := viewport.New(initialDimensionVal, initialDimensionVal)
+	viewportModel.MouseWheelEnabled = true
+
+	return &verboseState{
+		viewport:      viewportModel,
+		instanceTypes: instanceTypes,
+	}
+}
+
+// resizeVerboseView will change the dimensions of the verbose viewport in order to accommodate
+// the new window dimensions represented by the given tea.WindowSizeMsg
+func resizeVerboseView(model BubbleTeaModel, msg tea.WindowSizeMsg) BubbleTeaModel {
+	// This is needed to handle a bug with bubble tea
+	// where resizing causes misprints (https://github.com/Evertras/bubble-table/issues/121)
+	termenv.ClearScreen()
```

Comment:
Can we move this up right before we call all the component resizing functions so that we don't call this once for every component?

Reply:
We definitely can and should. Thanks for calling that out.